「招待URLをコピー」ボタンを作成する

uchihiro04 opened the pull request:
## 概要 
内見一覧ページに、「招待URLをコピー」ボタンを作成しました。
ボタンを押すと、HouseViewingレコードのuuidにもとづいたアプリ説明ページのURLがクリップボードにコピーされます。

## スクリーンショット
* 「招待URLをコピー」ボタン（赤枠にて強調）
<img width="700" alt="230706_招待URLをコピーボタン" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/fe0d2102-0615-4efc-b8d1-41ef15b8e83a">

デベロッパツールで「iPhone SE」の画面（375 × 667）を想定して表示した場合
<img width="200" alt="230706_招待URLをコピーボタン（スマートフォン）" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/0f94f50b-67b5-4d5f-bdbe-5717ad40bcd0">

* コピーしたURLでアプリ説明ページに遷移する様子

https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/64b65159-30f5-4227-9127-84286b5ca64a


## 関連Issue
- #90 

## 参考リンク
[Stimulus Handbook](https://stimulus.hotwired.dev/handbook/building-something-real)

## 備考
### 実装について
以下の実装は別Issueで行います。
* デザイン

Close #90

uchihiro04 committed:
#90 「招待URLをコピー」ボタンを作成

「招待URLをコピー」ボタンを作成しました。押すとアプリ説明ページに遷移するURLがクリップボードにコピーされます。

uchihiro04 committed:
#90 ヘルパーメソッドを作成

Slim-Lintから1行の文字数について指摘を受けたため、ヘルパーメソッドを作成しました。

uchihiro04 committed:
#90 targetsの要素名を変更

clipboard_controller.jsのtargetsについて、要素名を分かりやすくするためにsourceからurlに変更しました。

Yuma-Nagaoka commented on app/views/house_viewings/rooms/index.html.slim:
```diff
@@ -10,3 +10,6 @@ p 1件以上スコアの入力が完了していると、スコアを見るこ
 
 p URLを共有すると、複数人で採点ができます。
+.flex[data-controller="clipboard"]
+  input[data-clipboard-target="url" type="hidden" value="#{description_page_url(@house_viewing)}"]
```
細かいですが、`value="#{description_page_url(@house_viewing)}"`
これって`value=description_page_url(@house_viewing)`で行けませんでしたっけ

uchihiro04 replied:
確認したところ、ダブルクォーテーションで囲まなくても問題なく動作しました。ご指摘ありがとうございます。
レビューをもとに修正を行いました。該当するコミットは[6b03e5c](https://github.com/uchihiro04/house-viewing-score-log/pull/130/commits/6b03e5c5bedf26ea8f3802be62143d2dc87c4ba7)です。

Yuma-Nagaoka commented on app/javascript/controllers/clipboard_controller.js:
```diff
@@ -0,0 +1,9 @@
+export default class extends Controller {
+  static targets = [ "url" ]
+  
+  copyDescriptionPageUrl() {
```
関数名が長い気がしますしcopySharedUrlとかにしちゃいますか
意味的にも共有urlの方がしっくりくる気がします。

uchihiro04 replied:
ご指摘いただいた通り、copyDescriptionPageUrl（説明ページのURLのコピー）よりもcopySharedUrl（共有URLのコピー）のほうが、意図が伝わりやすいと感じました。
いただいたレビューをもとに修正を行いました。該当するコミットは[c1ec4d5](https://github.com/uchihiro04/house-viewing-score-log/pull/130/commits/c1ec4d55a743a6f73a567c7de2c7ed6167ea7350)です。

Yuma-Nagaoka commented on app/helpers/house_viewings/rooms_helper.rb:
```diff
@@ -0,0 +1,9 @@
+
+module HouseViewings
+  module RoomsHelper
+    def description_page_url(house_viewing)
```
合わせてこちらもshared_urlとかに

uchihiro04 replied:
ヘルパーメソッドを使う必要がなくなりましたため、description_page_urlメソッドも合わせて削除しています。該当するコミットは[6b03e5c](https://github.com/uchihiro04/house-viewing-score-log/pull/130/commits/6b03e5c5bedf26ea8f3802be62143d2dc87c4ba7)です。

Yuma-Nagaoka commented on app/helpers/house_viewings/rooms_helper.rb:
```diff
@@ -0,0 +1,9 @@
+module HouseViewings
+  module RoomsHelper
+    def description_page_url(house_viewing)
+      "#{request.base_url}/description/house_viewings/#{house_viewing.uuid}"
```
urlヘルパーメソッドで行けそうですね

uchihiro04 replied:
夕会でも確認しました通り、urlヘルパーメソッドを使用する形でも同じようにURLをコピーすることができました。
いただいたレビューをもとに修正を行いました。該当するコミットは[6b03e5c](https://github.com/uchihiro04/house-viewing-score-log/pull/130/commits/6b03e5c5bedf26ea8f3802be62143d2dc87c4ba7)です。

uchihiro04 committed:
#90 valueの中身をルーティングヘルパーを使用する形に変更

inputタグのvalueについて、ルーティングヘルパーを使用した形に変更しました。

uchihiro04 committed:
#90 メソッド名を変更

clipboard_controller.jsのメソッドcopyDescriptionPageUrlについて、メソッド名をcopySharedUrlに変更しました。

takuminasu reviewed (approved):
LGTM
